project/02-Modularity/Handout.md

```diff
@@ -0,0 +1,160 @@
+# Android Project - Part 2: Modularity
+
+In this second part of the SwEng project, you will add weather forecast capabilities to your app. Using an external service, [OpenWeatherMap](https://api.openweathermap.org/), you will display to the user a weather forecast at their location, or at an address of their choice. For that, you will also need to be able to find out where the user is located.
+
+> :information_source: In this part, we will use features that were introduced in Java 8. These features were not available on older versions of Android (Android < N - API Level 24). To use them, we either need to provide alternatives, or to loose compatibility with these older devices. Here, we choose to do the latter. To enable this, please go to your app `build.gradle` and set `minSdkVersion` to `24`.
+
+Today, we will mostly focus on writing good abstractions and working modules. Handling errors won't be part of this project, so it doesn't matter if your app crashes when something goes wrong. We will learn later how to handle errors gracefully.
+
+> :information_source: **This part of the project is rather long**. Next part will be way shorter, so you can spend about 1.5 weeks on this part.
+
+## Register to OpenWeatherMap
+
+[OpenWeatherMap](https://api.openweathermap.org/) offers an API that you can access over HTTPS. As this API offers both free and paid plans, they need to be able to identify the API caller, i.e. the App that is making the HTTPS requests. This is done using a secret key, that you provide in your requests to the service, and that identifies you uniquely.
```

Comment:
This is done using a secret key, that you provide in your requests to the service, and that identifies you uniquely.
->
This is done using a secret key, that identifies you uniquely, which you need to provide in your requests to the service.